test/viewmodels/SecurityStatementsViewModelSpec.scala

```diff
@@ -0,0 +1,142 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package viewmodels
+
+import config.AppConfig
+import models.FileRole.SecurityStatement
+import models.{EoriHistory, SecurityStatementsByPeriod, SecurityStatementsForEori}
+import org.scalatest.matchers.must.Matchers.convertToAnyMustWrapper
+import play.api.Application
+import play.api.i18n.Messages
+import play.twirl.api.{Html, HtmlFormat}
+import utils.CommonTestData.DAY_28
+import utils.SpecBase
+import utils.Utils.{
+  ddComponent, divComponent, dlComponent, dtComponent, h1Component,
+  linkComponent, pComponent, spanComponent
+}
+import views.html.components.{h2Inner, h3Inner, missing_documents_guidance, pInner, requestedStatements}
+
+import java.time.LocalDate
+
+class SecurityStatementsViewModelSpec extends SpecBase {
+
+  "SecurityStatementsViewModel" should {
+
+    "produce SecurityStatementsViewModel with correct contents" when {
+
+      "statementsForAllEoris has requested and current statements" in new Setup {
+        result.pageTitle mustBe Some(message("cf.security-statements.title"))
+        result.backLink mustBe Some(appConfig.customsFinancialsFrontendHomepage)
+        result.header mustBe expectedHeader
+        result.requestedStatementNotification mustBe expectedRequestedNotification
+        result.currentStatements mustBe expectedCurrentStatements
+        result.missingGuidance mustBe expectedMissingGuidance
+        result.statementServiceParagraph mustBe expectedStatementServiceParagraph
+        result.requestStatementsLink mustBe expectedRequestStatementsLink
+        result.details.hasRequestedStatements mustBe true
+        result.details.hasCurrentStatements mustBe true
+      }
+    }
+  }
+
+  trait Setup {
+    implicit val appConfig: AppConfig = mock[AppConfig]
+    val app: Application = application().build()
+    implicit val message: Messages = messages(app)
+
+    val date: LocalDate = LocalDate.now().withDayOfMonth(DAY_28)
+
+    val currentStatement: SecurityStatementsByPeriod = SecurityStatementsByPeriod(
+      startDate = date,
+      endDate = date,
+      files = Seq())
+
+    val requestedStatement: SecurityStatementsByPeriod = SecurityStatementsByPeriod(
+      startDate = date,
+      endDate = date,
+      files = Seq())
+
+    when(appConfig.customsFinancialsFrontendHomepage).thenReturn("home-page")
+    when(appConfig.historicRequestUrl(SecurityStatement)).thenReturn("request-statement-url")
+    when(appConfig.requestedStatements(SecurityStatement)).thenReturn("request-statement")
+
+    val expectedHeader: HtmlFormat.Appendable =
+      h1Component(msg = "cf.security-statements.title", classes = "govuk-heading-xl")
+
+    val expectedRequestedNotification: HtmlFormat.Appendable =
+      new requestedStatements(linkComponent).apply("request-statement")
+
+    val expectedCurrentStatements: HtmlFormat.Appendable = HtmlFormat.fill(Seq(
```

Comment:
Better to refactor this as it has become very hard to read as it has nested a lot. Better to create dl components outside of this and use it here. If you want to use dl component directly here then create dlComponent's content value outside of this. Will discuss this with you.

Reply:
This has been addressed in the commit [73dfe78](https://github.com/hmrc/customs-financials-documents-frontend/pull/101/commits/73dfe781532fd8bddb0967c9eeb15974440beaf7)